### DISCUSSION TITLE
New Images

### DISCUSSION THREAD

**Discussion by ResistanceIsUseless** [pry0cc/axiom#418] [Ideas]
Right now we have:
- Barebones Image
- Default
- Light
- ReconFTW 

I wonder if the following renaming/reconfiguring makes sense:
- Barebones -> Rename to Ubuntu base
- Default -> Rename to Fleet Image (Clean up apt commands and try to only have tools with modules)
- Copy of Default as it is today -> Keep as default or rename to Axiom Loaded
- Light (see no reason to change)
- ReconFTW -> Rename to Automation (Utilize containers and local installs to quickly spin up BB automation tools)

**Comment by 0xtavian** [comment]
Just some thoughts:
instead of having one big Packer file with all the tools, i was thinking of making a Packer JSON for each and every tool, and then during setup and depending on what u want installed, it will combine all the packer JSONs into one, kinda like chaining.

 the barebones JSON packer needs to be redone to be a minimal as possible with a baseline of security. So that build would still be called barebones. 

Then another build would be called controller, which is basically the  barebones + the install of axiom-configure spun up on a cloud hosted box. 

then another build, lets call it default would be the barebones + all the different JSONs for the tools to run most modules. 

then another called full, which would be barebones + default + all extra tools.

**Reply by 0xtavian** [reply]
the new barebones has been pushed to master, the easy part is done.

**Reply by 0xtavian** [reply]
@ResistanceIsUseless after looking into this work more, it seems the best way to chain Packer templates is by moving from legacy Packer JSONs to their new HCL2 templates. This is going to require all new Packer builders and provisioner files. The old JSONs are easily convertible to their new HCL2 template counterparts, as long as the builder is in the JSON. To see what a new HCL2 template looks like, update packer to 1.7.4 and supply a complete JSON with the following command `packer hcl2_upgrade .axiom/images/axiom.json`. Supplying only the [provisioners files](https://github.com/pry0cc/axiom/tree/master/images/provisioners) to convert will not work by default because during axiom-build we combine the [cloud builder](https://github.com/pry0cc/axiom/tree/master/images/builders) with the provisioner file, but Packer 1.7.x requires them to be together to convert. During axiom-build we store the final packer JSON in `.axiom/images/axiom.json`.  This is the converted JSON template of barebones for reference. 

[barebones.pkr.hcl.txt](https://github.com/pry0cc/axiom/files/7001586/barebones.pkr.hcl.txt)

edit: since our default.json has a Cloud Builder, auto converting works by default.
` packer hcl2_upgrade .axiom/images/provisioners/default.json`
[default.json.pkr.hcl.txt](https://github.com/pry0cc/axiom/files/7002488/default.json.pkr.hcl.txt)

**Comment by 0xtavian** [comment]
I know this is old but just to keep it up to date, this has been completed https://github.com/pry0cc/axiom/wiki/A-Quickstart-Guide#axiom-build @ResistanceIsUseless
